ckanext/scheming/plugins.py

```diff
@@ -234,28 +220,44 @@ def validate(self, context, data_dict, schema, action):
         thing, action_type = action.split('_')
         t = data_dict.get('type')
         if not t or t not in self._schemas:
-            return data_dict, {'type': [
-                "Unsupported dataset type: {t}".format(t=t)]}
+            return data_dict, {
+                'type': [
+                    "Unsupported dataset type: {t}".format(t=t)
+                ]
+            }
+
         scheming_schema = self._expanded_schemas[t]
 
-        if action_type == 'show':
-            get_validators = _field_output_validators
-        elif action_type == 'create':
-            get_validators = _field_create_validators
-        else:
-            get_validators = _field_validators
+        get_validators = {
+            'show': _field_output_validators,
+            'create': _field_create_validators,
+            'upate': _field_create_validators
```

Comment:
`upate`?  Also update doesn't use the `_field_create_validators`. I prefer the original code here even if it's longer.